lib/grandpa/message_handler_test.go

```diff
@@ -587,3 +587,251 @@ func TestMessageHandler_VerifyBlockJustification(t *testing.T) {
 	err = gs.VerifyBlockJustification(testHash, data)
 	require.Equal(t, ErrMinVotesNotMet, err)
 }
+
+func Test_getEquivocatoryVoters(t *testing.T) {
+	// many of equivocatory votes
+	ed25519Keyring, _ := keystore.NewEd25519Keyring()
+	fakeAuthorities := make([]*ed25519.Keypair, 0)
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Alice().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Alice().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Alice().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Bob().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Charlie().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Charlie().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Charlie().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Dave().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Dave().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Eve().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Ferdie().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Heather().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Heather().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Ian().(*ed25519.Keypair))
+	fakeAuthorities = append(fakeAuthorities, ed25519Keyring.Ian().(*ed25519.Keypair))
+
+	authData := make([]AuthData, len(fakeAuthorities))
+
+	for i, auth := range fakeAuthorities {
+		authData[i] = AuthData{
+			AuthorityID: auth.Public().(*ed25519.PublicKey).AsBytes(),
+		}
+	}
+
+	eqv := getEquivocatoryVoters(authData)
+	require.Len(t, eqv, 5)
+}
+
+func Test_VerifyCommitMessageJustification_ShouldRemoveEquivocatoryVotes(t *testing.T) {
+	t.Parallel()
+
```

Comment:
Let's avoid using `t.Parallel()` for now until we fix existing data races in the `lib/grandpa` package.

```suggestion
```